examples/vnet/doca_gw_dpdk.c

```diff
@@ -488,11 +530,29 @@ doca_gw_dpdk_build_vxlan_header(uint8_t **header,
 	*header += sizeof(vxlan_hdr);
 }
 
+static void
+doca_gw_dpdk_build_gre_header(uint8_t **header,
+		struct doca_gw_pipeline_cfg *cfg)
+{
+	uint32_t *key_data;
+	struct rte_gre_hdr gre_hdr;
+
+	memset(&gre_hdr, 0, sizeof(struct rte_gre_hdr));
+	gre_hdr.k = 1;
+	gre_hdr.proto = rte_cpu_to_be_16(RTE_ETHER_TYPE_IPV4);
```

Comment:
how do you know the next proto here? it might be ipv6 in the future, you need to get this info some how and fail if ipv6 not supported (until implemented)

Reply:
using match->inner ip type .done